git/server/service.go

```diff
@@ -59,3 +66,40 @@ func (g *gitService) Execute(dir string, command string, output io.Writer) error
 	}
 	return nil
 }
+
+func (g *gitService) isAllowedDirectory(directory string) bool {
+	for _, d := range g.allowedDirectories {
+		if directory == d {
+			return true
+		}
+	}
+	return false
+}
+
+func (g *gitService) isAllowedCommand(command string) bool {
+
+	if len(command) == 0 {
+		return false
+	}
+
+	commandFields := strings.Fields(command)
+
+	if commandFields[0] != "git" {
+		return false
+	}
+
+	for _, cmd := range g.allowedCommands {
+		allowedCmdFields := strings.Fields(cmd)
+		if len(commandFields) == len(allowedCmdFields) {
+			valid := true
+			for i, cmdAllowedField := range allowedCmdFields {
+				valid = valid && (cmdAllowedField == commandFields[i])
+			}
+			if valid {
```

Comment:
Because I don't want it to return all the time, only when valid is true. In that way the loop can continue trying to match with the remaining "allowedCommands". Not sure if there's a different way of doing it :)